docs/proposals/33-machine-image-capabilities.md

```diff
@@ -0,0 +1,639 @@
+- Minimize the additional data size introduced to cloud profiles to prevent bloating.
+- Obsolete existing workarounds described in the "Current Workarounds" section.
+
+### Non-Goals
+
+- Introduce capability options that are not utilized by Gardener in the future.
+- Introduce feature flag to control the usage of a capability, e.g. secure boot.
+
+## Proposal
+
+Introduce a top level capabilities map in the CloudProfile `spec.capabilities`.
+
+Capabilities are very specific to the provider and the selected catalog offered by Gardener in the cloud profile.
+To minimize complexity and data size in the cloud profile, the capabilities are defined as a map with string keys and string arrays as values.
+The key is the capability name and the value is an array of possible values.
+
+```go
+type Spec struct {
+    Capabilities map[string][]string `json:"capabilities"`
+}
+```
+
+For each cloud profile the capabilities are defined in the `spec.capabilities` map.
+The full set of possibilities for each capability is defined here.
+As some capabilities can have multiple values at the same time an array of possible values is used instead of a single value.
+
+This structure defines the default set of capabilities.
+If no further information are provided each machine type and machine image will be assigned the default set of capabilities.
+Thus being compatible with each other.
+
+```yaml
+# CloudProfile Example
+spec:
+  capabilities:
+    hypervisorType: ["gen2", "gen1"]
+    network: ["accelerated", "standard"]
+    storageAccess: ["NVMe", "SCSI"]
+    secureBoot: ["secure", "none"]
+    bootMode: ["uefi-preferred", "uefi", "legacy-bios"]
+    ...
+```
+
+The array of a capability is ordered.
+The order defines the priority of the values in case of multiple supported images.
+The first value in the array is the most preferred value.
+The last value is the least preferred value.
+
+Example: A machine supports hypervisor `gen2` AND `gen1` and an image version offers `gen1` OR `gen2`.
+Then the image with `gen2` will be preferred.
+
+Other edge cases will be deterministic solved as well to ensure a consistent behavior.
+
+In addition to the default capabilities, the `spec.machineTypes` is extended with the `capabilities` structure described above.
+The image versions in `spec.machineImages.versions` will be extended with `capabilityCombinations` - an array of capabilities structures, one entry for each image reference of an image version in the cloud provider.
+
+The architecture is also added to the capabilityCombinations. This is required as the architecture is a capability itself. 
+
+❗️❗️❗️NOTE: To avoid duplication and remove the possibility of inconsistencies the existing `architecture` field on `machineImages.versions` and `machineTypes` will be marked as deprecated. Architecture will become the only mandatory capability and will take precedence over the deprecated field if both are present.
```

Comment:
```suggestion
> [!NOTE]
> To avoid duplication and remove the possibility of inconsistencies the existing `architecture` field in `machineImages.versions` and `machineTypes` will be marked as deprecated. `architecture` will become the only mandatory capability and will take precedence over the deprecated field if both are present.
```